java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -79,6 +87,12 @@ private static boolean hasOnlyStaticMembers(ClassTree classTree) {
     return true;
   }
 
+  private static boolean isSpringBootApplication(ClassTree classTree) {
+    SpringBootVisitor springBootVisitor = new SpringBootVisitor();
```

Comment:
I am not a found of this subvisitor... but other solution are worst IMO : We could approach things by using leavenode on classTree and stack of boolean and visit Class and MethodInvocation. But i tend to think that'll make things more complicated. So maybe keep thing as is for now.